parsl/executors/workqueue/executor.py

```diff
@@ -329,19 +329,21 @@ def submit(self, func, resource_specification, *args, **kwargs):
 
             acceptable_resource_types = ['cores', 'memory', 'disk']
             keys = list(resource_specification.keys())
-            if len(keys) != 3:
-                logger.error("Task resource specification requires "
-                             "three resources to be specified simultaneously: cores, memory, and disk")
-                raise ExecutorError(self, "Task resource specification requires "
-                                          "three resources to be specified simultaneously: cores, memory, and disk, "
-                                          "and only takes these three resource types.")
-
             if not all(k.lower() in acceptable_resource_types for k in keys):
                 logger.error("Task resource specification only accepts "
                              "three types of resources: cores, memory, and disk")
-                raise ExecutorError(self, "Task resource specification requires "
-                                          "three resources to be specified simultaneously: cores, memory, and disk, "
-                                          "and only takes these three resource types.")
+                raise ExecutorError(self, "Task resource specification only accepts "
+                                          "three types of resources: cores, memory, and disk")
+
+            if len(keys) != 3:
```

Comment:
Might make sense to convert this to set comparisons to make this a bit more clearer.